Py_FS/wrapper/nature_inspired/ALO.py

```diff
@@ -0,0 +1,146 @@
+"""
+Programmer: Rishav Pramanik
+Date of Development: 20/08/2021
+Antlion Optimizer
+"""
+import numpy as np
+from wrapper.nature_inspired.algorithm import Algorithm
+from wrapper.nature_inspired._transfer_functions import get_trans_function
+from sklearn import datasets
+
+class ALO(Algorithm):
+    def __init__(self,
+                 num_agents,
+                 max_iter,
+                 train_data,
+                 train_label,
+                 test_data=None,
+                 test_label=None,
+                 val_size=20,
+                 trans_func_shape='s',
+                 seed=0,
+                 save_conv_graph=True,
+                 max_evals=np.float("inf")):
+        self.trans_func_shape=trans_func_shape
+        super().__init__( num_agents=num_agents,
+                        max_iter=max_iter,
+                        train_data=train_data,
+                        train_label=train_label,
+                        save_conv_graph=save_conv_graph,
+                        seed=seed )
+
+        self.algo_name = 'ALO'
+        self.agent_name = 'Antlion'
+        self.trans_function = trans_func_shape
+        self.algo_params = {}
+        self.popul = initialize(self.num_agents, self.train_data.shape[-1]) #initialise the ants
```

Comment:
What is self.popul?

Reply:
It is the ants

Reply:
@rishavpramanik contradiction while calling combine function in line 127.

Reply:
The variables are a bit confusing. If the name for ants is population in the original paper, then instead change self.popul to self.ants

Reply:
Updated